### DISCUSSION TITLE
Spring kafka 2.8 - adding dynamic micrometer tag from record header

### DISCUSSION THREAD

**Discussion by erohana** [spring-projects/spring-kafka#2632] [Q&A]
Hi,
i'm looking for a way to enrich the `spring.kafka.listener` metric with metadata from the consumer record.
basically, I'm looking for a way to do it like  [WebMvcTagsContributor](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/actuate/metrics/web/servlet/WebMvcTagsContributor.html)

is there way to do this customization ?

**Comment by artembilan** [comment]
Well, looks like you are pointing to the deprecated API.
See its Javadocs carefully:
```
Deprecated, for removal: This API element is subject to removal in a future version.
since 3.0.0 for removal in 3.2.0 in favor of [ServerRequestObservationConvention](https://docs.spring.io/spring-framework/docs/6.0.6/javadoc-api/org/springframework/http/server/observation/ServerRequestObservationConvention.html)
```
From here it must not be so hard to follow Spring for Apache Kafka docs and realize that there is something similar with a `KafkaListenerObservationConvention`: https://docs.spring.io/spring-kafka/reference/html/#observation.

Does that make sense to you?

**Reply by erohana** [reply]
but this is from spring 6.0 version and spring boot 3
I'm working with spring boot 2.7 and spring kafka 2.8

is there a way to do it in the old version (2.8)?

**Reply by garyrussell** [reply]
2.8 is no longer supported; use 2.9.x with Boot 2.7.x. https://spring.io/projects/spring-kafka#support

However, the answer is no; not currently; only static tags can be added via `ContainerProperties`...

```java
	/**
	 * Set additional tags for the Micrometer listener timers.
	 * @param tags the tags.
	 * @since 2.3
	 */
	public void setMicrometerTags(Map<String, String> tags) {
```

You would need to create your own meters within your listener.

**Reply by erohana** [reply]
ok thanks for answering

**Reply by garyrussell** [reply]
https://github.com/spring-projects/spring-kafka/issues/2638
